circle/functions

```diff
@@ -581,28 +604,120 @@ update_chart_in_repo() {
-    do
-      log "Deleting $branch..."
-      git push development :$branch >/dev/null
-    done
+    git_cleanup_old_branches "${CHART_NAME}-*" "^$CHART_NAME-$CHART_VERSION_FIRST_UPDATE$"
   else
     warn "Chart release/updates skipped!"
   fi
-  
+
   info "Cleaning up repository"
   # Exiting from charts/ repository
   cd ..
   rm -rf charts
 }
+
+# Expects three parameters: <repo to update> <distro codename> and <docker image digest>
+update_minideb_derived() {
+    if [ -z "$1" ] || [ -z "$2" ] || [ -z "$3" ]
+    then
+        error "Wrong arguments passed to update_minideb_extras function"
+        exit 1
+    fi
+
+    REPO_TO_UPDATE=$1
+    DIST=$2
+    DIST_BUILD_ID=$3
+    BASENAME=bitnami/minideb
+
+    echo "Cloning '$REPO_TO_UPDATE' repo..."
+    if ! git clone --quiet --single-branch $REPO_TO_UPDATE minideb-extras; then
+        error "Could not clone $REPO_TO_UPDATE..."
+        exit 1
+    fi
+    cd minideb-extras
+
+    # add development remote
+    git remote add development "https://$GITHUB_USER@github.com/$GITHUB_USER/$(echo ${REPO_TO_UPDATE/https:\/\/github.com\/bitnami\/} | tr / -).git"
+
+    # lookup the distro in minideb-extras
+    DIST_PATH=
+    for d in $(find * -type d -name $DIST )
+    do
+        if [ -f "$d/Dockerfile" ]; then
+            DIST_PATH=$d
+            break
+        fi
+    done
+
+    if [[ -z $DIST_PATH ]]; then
+        error "Distro '$DIST' could not be found in '$REPO_TO_UPDATE' repo"
+        continue
+    else
+        cd "$DIST_PATH"
+    fi
+
+    if [[ -z $GITHUB_USER || -z $GITHUB_PASSWORD ]]; then
+        error "GitHub credentials not configured. Aborting..."
+        exit 1
+    fi
+
+    git_configure
+
+    # Check if we need to update the Dockerfile
+    if grep "${DIST_BUILD_ID}" Dockerfile &>/dev/null ; then
+        echo "Dockerfile for dist ${DIST} is already using the new build"
+        continue
+    fi
+
+    # Create branch
+    BRANCH_NAME_BASE="minideb_update"
+    BRANCH_NAME="${BRANCH_NAME_BASE}_${DIST}_$(date +%Y%m%d_%H%M)"
+    if ! git_create_branch $BRANCH_NAME ; then error "Could not create new branch..."; exit 1; fi
+
+    # Substitute and bump image version
+    sed -i "s|^FROM .*$|FROM ${BASENAME}@${DIST_BUILD_ID}|" ./Dockerfile
+    # Bump image version
+    IMAGE_VERSION=$(grep '^ENV BITNAMI_IMAGE_VERSION' ./Dockerfile | awk -F '=' '{print $2}')
+    NEW_IMAGE_VERSION=${IMAGE_VERSION%r*}r$((${IMAGE_VERSION##*r}+1))
+    sed -i "s|^ENV BITNAMI_IMAGE_VERSION.*$|ENV BITNAMI_IMAGE_VERSION=${NEW_IMAGE_VERSION}|" ./Dockerfile
+
+    # Commit changes
+    git add ./Dockerfile
+    git commit -m "[$DIST] Update minideb base image" >/dev/null
+
+    # Push branch and create pull request
+    echo "Publishing branch to remote repo..."
+    git push development ${BRANCH_NAME} >/dev/null
+
+    install_hub || exit 1
+
+    echo "Creating pull request with '$REPO_TO_UPDATE' repo..."
+    if ! hub pull-request -m "[$DIST] Update minideb base image"; then
```

Comment:
this will fail if there already is a pull request that was not merged and branch was not deleted. 
`git_create_branch` will define `BRANCH_AMEND_COMMITS=1` if the branch already exists (ref: https://github.com/bitnami/test-infra/blob/master/circle/functions#L395). You can use this to decide whether you should create a new PR as is done here https://github.com/bitnami/test-infra/blob/master/circle/functions#L573.

Reply:
This process will be automatic so there wont be any PR to amend commits into